packages/stable/src/__tests__/api/documentsApi.unit.spec.ts

```diff
@@ -0,0 +1,293 @@
+// Copyright 2022 Cognite AS
+
+import nock from 'nock';
+import CogniteClient from '../../cogniteClient';
+import { setupMockableClient } from '../testUtils';
+import { mockBaseUrl } from '../testUtils';
+
+describe('Documents unit test', () => {
+  let client: CogniteClient;
+  beforeEach(() => {
+    client = setupMockableClient();
+    nock.cleanAll();
+  });
+
+  test('search with query', async () => {
+    nock(mockBaseUrl)
+      .post(new RegExp('/documents/search'), {
+        search: {
+          query: 'test',
+        },
+      })
+      .once()
+      .reply(200, {});
+    await client.documents.search({
+      search: { query: 'test' },
+    });
+  });
+
+  describe('geo location', () => {
+    test('point', async () => {
+      nock(mockBaseUrl)
+        .post(new RegExp('/documents/search'), {})
+        .once()
+        .reply(200, {
+          items: [
+            {
+              item: {
+                geoLocation: {
+                  type: 'Point',
+                  coordinates: [2.324, 23.1],
+                },
+              },
+            },
+          ],
+        });
+      const response = await client.documents.search({});
+      const geoLocation = response.items[0].item.geoLocation;
+      expect(geoLocation?.type).toEqual('Point');
+      expect(geoLocation?.coordinates).toBeDefined();
+      // @ts-ignore
```

Comment:
The cause for this was simply that the `coordinates` field is optional. I added a `!` behind it, like so:

```js
expect(geoLocation?.coordinates![0]).toEqual(2.324);
```

This removes the need for the `@ts-ignore`

I have not done anything about the `any[]`. Looking at the implementation in the `unstructured-search` repo, it looks like this depends a bit on the type of shape, so it'll get a bit complex if we want to do it exact. I think the `any[]` here is about on par with how it's defined in the contract.

Reply:
The `any[]` is aligned with the service spec, but I think we should look later to see if we can improve this (in the service spec, and also here). Currently a client have _no idea_ what it might contain (as it is defined as anything) without looking into external resources, while it actually can only hold

```ts
type Position = [number, number]
type Coordinates =
  | Position // Point
  | Position[] // MultiPoint, ListString
  | Position[][] // MultiLineString, Polygon
  | Position[][][] // MultiPolygon
```

This would narrow everything down to numbers and a few lists, and making things a lot simpler to reason about.